src/rsa/padding.rs

```diff
@@ -164,9 +164,79 @@ pub struct PSS {
     pub digest_alg: &'static digest::Algorithm,
 }
 
-// Maxmimum supported output length for PSS padding.
+#[cfg(feature = "rsa_signing")]
+// Maximum supported length of the salt in bytes.
+// In practice, this is constrained by the maximum digest length.
+const MAX_SALT_LEN: usize = 512 / 8;
+
+// Maxmimum supported output length for PSS padding in bytes.
 const MAX_OUTPUT_LEN: usize = 4096;
 
+#[cfg(feature = "rsa_signing")]
+impl Encoding for PSS {
+    // Implement padding procedure per EMSA-PSS,
+    // https://tools.ietf.org/html/rfc3447#section-9.1.
+    fn encode(&self, msg: &[u8], out: &mut [u8], rng: &rand::SecureRandom)
+              -> Result<(), ()> {
+
```

Comment:
In all your commits, please remove this blank line at the top of each function where one exists.